### PR TITLE
Add timeout feature

### PR DESCRIPTION
**Pull request by akshaysarraf2016**
Add timeout feature

### CODE REVIEW

**Comment by allburov** [comment]
@akshaysarraf2016 good work, I appreciate it! 
Could you add a little note to the README.md how to use `timeout` with examples? The best way I guess give a few examples and add a link to the original `requests.timeout` documentation

**Comment by allburov** [comment]
Have you tested it and does it work, btw? I dont have an artifactory instance now and can't test it :(

**Comment by akshaysarraf2016** [comment]
I've updated README.md
Yes, It works. 
This exception is thrown when a requests is timed out:


`Traceback (most recent call last):
  File "C:\Python38\lib\site-packages\urllib3\connectionpool.py", line 421, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "C:\Python38\lib\site-packages\urllib3\connectionpool.py", line 416, in _make_request
    httplib_response = conn.getresponse()
  File "C:\Python38\lib\http\client.py", line 1347, in getresponse
    response.begin()
  File "C:\Python38\lib\http\client.py", line 307, in begin
    version, status, reason = self._read_status()
  File "C:\Python38\lib\http\client.py", line 268, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "C:\Python38\lib\socket.py", line 669, in readinto
    return self._sock.recv_into(b)
  File "C:\Python38\lib\ssl.py", line 1241, in recv_into
    return self.read(nbytes, buffer)
  File "C:\Python38\lib\ssl.py", line 1099, in read
    return self._sslobj.read(len, buffer)
socket.timeout: The read operation timed out

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Python38\lib\site-packages\requests\adapters.py", line 439, in send
    resp = conn.urlopen(
  File "C:\Python38\lib\site-packages\urllib3\connectionpool.py", line 719, in urlopen
    retries = retries.increment(
  File "C:\Python38\lib\site-packages\urllib3\util\retry.py", line 400, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "C:\Python38\lib\site-packages\urllib3\packages\six.py", line 735, in reraise
    raise value
  File "C:\Python38\lib\site-packages\urllib3\connectionpool.py", line 665, in urlopen
    httplib_response = self._make_request(
  File "C:\Python38\lib\site-packages\urllib3\connectionpool.py", line 423, in _make_request
    self._raise_timeout(err=e, url=url, timeout_value=read_timeout)
  File "C:\Python38\lib\site-packages\urllib3\connectionpool.py", line 330, in _raise_timeout
    raise ReadTimeoutError(
urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='XXXXXX', port=443): Read timed out. (read timeout=0.1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 6, in <module>
    with path.open() as fd:
  File "C:\work\CR_Codes\artifactory\artifactory\artifactory.py", line 1351, in open
    return self._accessor.open(self)
  File "C:\work\CR_Codes\artifactory\artifactory\artifactory.py", line 902, in open
    raw, code = self.rest_get_stream(
  File "C:\work\CR_Codes\artifactory\artifactory\artifactory.py", line 672, in rest_get_stream
    res = session.get(url, stream=True, verify=verify, cert=cert, timeout=timeout)
  File "C:\Python38\lib\site-packages\requests\sessions.py", line 543, in get
    return self.request('GET', url, **kwargs)
  File "C:\Python38\lib\site-packages\requests\sessions.py", line 530, in request
    resp = self.send(prep, **send_kwargs)
  File "C:\Python38\lib\site-packages\requests\sessions.py", line 643, in send
    r = adapter.send(request, **kwargs)
  File "C:\Python38\lib\site-packages\requests\adapters.py", line 529, in send
    raise ReadTimeout(e, request=request)
requests.exceptions.ReadTimeout: HTTPSConnectionPool(host='XXXXXX', port=443): Read timed out. (read timeout=0.1)`

File: README.md [hunk 1]
```diff
@@ -974,6 +975,32 @@ import requests.packages.urllib3 as urllib3
 urllib3.disable_warnings()
 ```
 
+## Timeout on requests ##
+
```
Comment (allburov) [inline README.md:979]:
```suggestion

The library supports `timeout` argument in the same meaner as [requests does](https://requests.readthedocs.io/en/master/user/advanced/#timeouts)
```

**Comment by allburov** [comment]
@akshaysarraf2016 thanks!

**Pull request merged**